src/content/docs/dropins-b2b/purchase-order/events.mdx

```diff
@@ -0,0 +1,712 @@
+      badge.textContent = count;
+      badge.style.display = count > 0 ? 'block' : 'none';
+    }
+  }
+  
+  viewOrder(orderNumber) {
+    window.location.href = `/purchase-orders/${orderNumber}`;
+  }
+  
+  async loadExistingOrders() {
+    // Load existing orders from API
+    const { pendingOrders, completedOrders } = await fetchPurchaseOrders();
+    this.pendingOrders = pendingOrders;
+    this.completedOrders = completedOrders;
+  }
+}
+
+// Initialize dashboard
+const dashboard = new PurchaseOrderDashboard();
+```
+
+
+
+#### Usage scenarios
+
+- Display success confirmation with order details.
+- Redirect to the success page (approval or direct order).
+- Clear shopping cart after successful placement.
+- Send email notifications to approvers.
+- Track analytics and conversion events.
+- Update purchase order history and counts.
+- Show real-time notifications for new orders.
+- Update dashboard widgets and badges.
+- Trigger approval workflow notifications.
+- Log purchase order creation for audit.
+- Update budget tracking systems.
+- Sync with ERP/accounting systems.
+
+---
+
+
+
+### `purchase-order/refresh` (emits and listens)
+
+Emitted and consumed for internal and external communication.
+
+#### Event payload
+
+```typescript
+Boolean
+```
+
+
+
+#### When triggered
+
+- After approval rule changes
+- After permission updates
+- On user request (manual refresh)
+- After significant state changes
+
+
+
+#### Example: Example
+
+```js
+import { events } from '@dropins/tools/event-bus.js';
+
+// Emit refresh event
+events.emit('purchase-order/refresh', {
+  purchaseOrderId: 'PO123456'
+});
+
+// Listen for refresh requests
+events.on('purchase-order/refresh', async (payload) => {
+  if (payload.data.purchaseOrderId) {
+    // Refresh specific purchase order
+    await refreshPurchaseOrder(payload.data.purchaseOrderId);
+  } else {
+    // Refresh all purchase orders
+    await refreshAllPurchaseOrders();
+  }
+});
+```
+
+
+
+#### Usage scenarios
+
+- Force reload after external updates.
+- Implement pull-to-refresh functionality.
+- Sync data after background changes.
+- Refresh after approval rule modifications.
+
+---
+
+## Listening to events
+
+All Purchase Order events are emitted through the centralized event bus. Subscribe to events using the `events.on()` method:
+
+```js
+import { events } from '@dropins/tools/event-bus.js';
+
+// Listen to purchase order lifecycle events
+events.on('purchase-order/placed', handlePurchaseOrderPlaced);
+events.on('purchase-order/data', handlePurchaseOrderData);
+events.on('purchase-order/refresh', handleRefreshRequest);
+
+// Listen to permission changes
+events.on('auth/permissions', handlePermissionUpdate);
+
+// Remove listeners when done
+events.off('purchase-order/placed', handlePurchaseOrderPlaced);
+```
+
+<Aside type="tip">
+The `purchase-order/data` event is frequently emitted. Consider debouncing or throttling your event handlers if they perform expensive operations.
```

Comment:
Seems a bit incorrect for me. This event usually emits only once per PO details page load, and then also might be emitted when something changed by user (e.g.: new comment posted or some action like PO approve/cancel done)